.github/scripts/download_triton_packages.py

```diff
@@ -0,0 +1,209 @@
+# This file is extracted from triton's setup.py file.
+# It allows us to download triton's dependencies without running the full build
+import os
+import platform
+import shutil
+import tarfile
+import urllib.request
+
+from pathlib import Path
+from subprocess import check_call
+from tempfile import TemporaryDirectory
+from typing import Any, List, NamedTuple
+
+SCRIPT_DIR = Path(__file__).parent
+REPO_DIR = SCRIPT_DIR.parent.parent
+
+
+class Package(NamedTuple):
+    package: str
+    name: str
+    url: str
+    include_flag: str
+    lib_flag: str
+    syspath_var_name: str
+
+
+# pybind11
+
+
+def get_pybind11_package_info() -> Package:
+    name = "pybind11-2.11.1"
+    url = "https://github.com/pybind/pybind11/archive/refs/tags/v2.11.1.tar.gz"
+    return Package(
+        "pybind11", name, url, "PYBIND11_INCLUDE_DIR", "", "PYBIND11_SYSPATH"
+    )
+
+
+# llvm
+
+
+def get_llvm_package_info_cuda() -> Package:
+    system = platform.system()
+    arch = platform.machine()
+    if arch == "aarch64":
+        arch = "arm64"
+    if system == "Darwin":
+        arch = platform.machine()
+        if arch == "x86_64":
+            arch = "x64"
+        system_suffix = f"macos-{arch}"
+    elif system == "Linux":
+        if arch == "arm64":
+            system_suffix = "ubuntu-arm64"
+        else:
+            vglibc = tuple(map(int, platform.libc_ver()[1].split(".")))
+            vglibc = vglibc[0] * 100 + vglibc[1]
+            system_suffix = "ubuntu-x64" if vglibc > 217 else "centos-x64"
+    else:
+        return Package(
+            "llvm",
+            "LLVM-C.lib",
+            "",
+            "LLVM_INCLUDE_DIRS",
+            "LLVM_LIBRARY_DIR",
+            "LLVM_SYSPATH",
+        )
+    # use_assert_enabled_llvm = check_env_flag("TRITON_USE_ASSERT_ENABLED_LLVM", "False")
+    # release_suffix = "assert" if use_assert_enabled_llvm else "release"
+    llvm_hash_file = open("../cmake/llvm-hash.txt")
+    rev = llvm_hash_file.read(8)
+
+    name = f"llvm-{rev}-{system_suffix}"
+    url = f"https://tritonlang.blob.core.windows.net/llvm-builds/{name}.tar.gz"
+    return Package(
+        "llvm", name, url, "LLVM_INCLUDE_DIRS", "LLVM_LIBRARY_DIR", "LLVM_SYSPATH"
+    )
+
+
+def get_llvm_package_info_rocm() -> Package:
+    system = platform.system()
+    arch = platform.machine()
+    if arch == "aarch64":
+        arch = "arm64"
+    if system == "Darwin":
+        system_suffix = "apple-darwin"
+        arch = platform.machine()
```

Comment:
This is all moot for rocm (as Linux only solution, isn't it?)

Reply:
I don't think this code is used in CI, no. However, I don't want to make too many changes from the original code in the rocm-triton repo.